tests/pgautofailover_utils.py

```diff
@@ -892,11 +892,13 @@ def has_needed_replication_slots(self):
         if set(expected_slots) == set(current_slots):
             print("slots list on %s is %s, as expected" %
                   (self.datadir, current_slots))
-        else:
-            print("slots list on %s is %s, expected %s" %
-                  (self.datadir, current_slots, expected_slots))
+            return True
 
-        return set(expected_slots) == set(current_slots)
+        self.print_debug_logs()
```

Comment:
I guess that's just for the development phase, not intended for commit?

Reply:
It is intended for commit. This is only printed when this check fails, so the logs are very useful for investigating.